kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.kinesis;
+
+import com.amazonaws.SDKGlobalConfiguration;
+import com.amazonaws.regions.Regions;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.config.ProcessingGuarantee;
+import com.hazelcast.jet.kinesis.KinesisSinks;
+import com.hazelcast.jet.kinesis.KinesisSources;
+import com.hazelcast.jet.kinesis.impl.AwsConfig;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sink;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.StreamSource;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.hazelcast.jet.core.JobStatus.FAILED;
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.tests.common.Util.entry;
+import static com.hazelcast.jet.tests.common.Util.getJobStatusWithRetry;
+import static com.hazelcast.jet.tests.common.Util.sleepMillis;
+import static com.hazelcast.jet.tests.common.Util.sleepMinutes;
+import static com.hazelcast.jet.tests.common.Util.waitForJobStatus;
+
+public class KinesisTest extends AbstractSoakTest {
+
+    private static final int DEFAULT_SHARD_COUNT = 1;
+    private static final int DEFAULT_PARTITION_KEYS = 10_000;
+    private static final int DEFAULT_SLEEP_MS_BETWEEN_ITEM = 1;
+    private static final int DEFAULT_SNAPSHOT_INTERVAL = 5000;
+
+    private final List<String> streamNames = new ArrayList<>(2);
+
+    private AwsConfig awsConfig;
+    private Helper helper;
+
+    private int partitionKeys;
+    private int shardCount;
+    private long sleepMsBetweenItem;
+    private long snapshotIntervalMs;
+
+    public static void main(String[] args) throws Exception {
+        System.setProperty(SDKGlobalConfiguration.AWS_CBOR_DISABLE_SYSTEM_PROPERTY, "true");
+        new KinesisTest().run(args);
+    }
+
+    @Override
+    protected void init(JetInstance client) {
+        awsConfig = new AwsConfig()
+                .withEndpoint(property("endpoint", "http://localhost:4566"))
+                .withRegion(Regions.US_EAST_1.getName())
```

Comment:
This doesn't seem to be configurable via properties.

Reply:
I've removed it intentionally, we run soak-tests on `us_east_1` and it is hardcoded. I thought it'll be the same for kinesis so removed the parameter.